Update _search_successive_halving.py

ggrrll opened the pull request:
#### Reference Issues/PRs
related to #24777 

#### What does this implement/fix? Explain your changes.
it fixes the error message

ggrrll committed:
Update _search_successive_halving.py

jeremiedbb commented on sklearn/model_selection/_search_successive_halving.py:
```diff
@@ -183,7 +183,7 @@ def _check_input_parameters(self, X, y, groups):
             if not self.resource == "n_samples":
                 raise ValueError(
-                    "max_resources can only be 'auto' if resource='n_samples'"
+                    "resource can only be 'n_samples' if max_resources='auto'"
```
small nitpick
```suggestion
                    "resource can only be 'n_samples' when max_resources='auto'"
```

jeremiedbb reviewed (commented):
Thanks @ggrrll. Just a small nitpick

jeremiedbb commented:
The CI is failing because we have a test that check the content of the error message. You'll need to update the expected error message there as well https://github.com/scikit-learn/scikit-learn/blob/02ebf9e68fe1fc7687d9e1047b9e465ae0fd945e/sklearn/model_selection/tests/test_successive_halving.py#L353.

ggrrll committed:
Update sklearn/model_selection/_search_successive_halving.py

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

ggrrll committed:
Merge pull request #1 from scikit-learn/main

DOC FIX Consistent formulae for metrics in the user guide (#24673)

ggrrll committed:
update tests

addressing https://github.com/scikit-learn/scikit-learn/pull/24781#issuecomment-1295001638

jeremiedbb commented on sklearn/model_selection/tests/test_successive_halving.py:
```diff
@@ -350,7 +350,7 @@ def test_random_search_discrete_distributions(
         (
             {"max_resources": "auto", "resource": "b"},
-            "max_resources can only be 'auto' if resource='n_samples'",
+             "resource can only be 'n_samples' when max_resources='auto'"
```
```suggestion
            "resource can only be 'n_samples' when max_resources='auto'",
```

jeremiedbb committed:
Update sklearn/model_selection/tests/test_successive_halving.py

jeremiedbb reviewed (approved):
Thanks @ggrrll
